miden-tx/src/prover/mod.rs

```diff
@@ -61,6 +65,18 @@ impl TransactionProver {
             TransactionKernel::from_transaction_parts(&stack_outputs, &map.into(), output_notes)
                 .map_err(TransactionProverError::InvalidTransactionOutput)?;
 
+        // erase private note information (convert private full notes to just headers)
+        let output_notes: Vec<OutputNote> = tx_outputs
+            .output_notes
+            .iter()
+            .map(|note| match note {
+                OutputNote::Full(note) if note.metadata().note_type() == NoteType::OffChain => {
+                    OutputNote::Header(*note.header())
+                },
+                _ => note.clone(),
```

Comment:
I wonder if this logic should go into `OutputNote` struct. For example, we could have something like:
```Rust
pub fn shrink(&self) -> Self {
    match self {
        Self::Full(note) if note.isOffChain() => Self::Header(*note.header()),
        _ => self.clone(),
    }
}
```